backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/work/impl/CreateIndexWork.java

```diff
@@ -94,8 +94,11 @@ public Builder settings(IndexSettings settings, IndexSettings customSettings) {
 			 * We better not include the null fields.
 			 */
 			Gson gson = gsonProvider.getGsonNoSerializeNulls();
-			// TODO HSEARCH-3934 Merge customSettings into settings
-			payload.add( "settings", gson.toJsonTree( ( customSettings == null ) ? settings : customSettings ) );
+
+			// if customSettings are present, merge them with the ones created by Search
+			settings.merge( customSettings );
```

Comment:
Since this is a mutating method, it would be better to execute it during metadata building, don't you think? It feels a bit weird to alter the metadata stored in the `IndexManager` just because someone requested to create the index in the Elasticsearch cluster.

Reply:
right! it is better to merge them earlier